flow/compositor_context.h

```diff
@@ -37,6 +37,15 @@ enum class RasterStatus {
 
 class CompositorContext {
  public:
+  class Delegate {
+   public:
+    /// Called at the end of a frame with approximately how many bytes mightbe
+    /// freed if a GC ran now.
+    virtual void OnCompositorEndFrame(size_t freed_hint) = 0;
```

Comment:
Please document the threading restrictions that implementors must adhere to. I believe the shell is not thread-unsafe. Details below.

Reply:
I'm not sure what thread restrictions to document here. This is just called when there's a hint available for bytes freed - shouldn't it be up to the implementer what threads it uses when processing that information?

Reply:
I meant whoever implements the delegate protocol will receive events on the raster thread. This might be unexpected to the implementer.